File: leetcode.com/0129_sum_root_to_leaf_numbers/solution.go
func dfs(node *TreeNode, total int) int {
    if node.Left == nil && node.Right == nil {
        return total*10 + node.Val
    }

    v := 0

    if node.Left != nil {
        v += dfs(node.Left, total*10 + node.Val)
    }

    if node.Right != nil {
        v += dfs(node.Right, total*10 + node.Val)
    }

    return v
}

func sumNumbers(root *TreeNode) int {
    return dfs(root, 0)
}

File: leetcode.com/0129_sum_root_to_leaf_numbers/solution.py
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:

        def dfs(node, path):
            if node.left is node.right is None:
                return int(''.join(map(str, path + [node.val])))

            total = 0

            if node.left:
                total += dfs(node.left, path[:] + [node.val])

            if node.right:
                total += dfs(node.right, path[:] + [node.val])

            return total

        return dfs(root, [])
